### PR TITLE
Update catalog docs with new api

### PR DESCRIPTION
**Pull request by ah-**
I'm just reading through the docs and noticed a few occasions where the examples don't work anymore.

The new versions are just my best guess and what seems to work, please update if there's a better way of doing this.

### CODE REVIEW

File: docs/source/catalog.rst [hunk 1]
```diff
@@ -134,11 +134,11 @@ A Catalog can be loaded from a YAML file on the local filesystem by creating a C
 
 Then sources can be listed::
 
-    cat.list()
+    list(cat)
 
 and data sources are loaded with ``get()``::
 
-    data = cat.get('entry1_part', part='1')
+    data = cat['entry_part1'].get(part='1')
```
Comment (martindurant) [inline docs/source/catalog.rst]: Does this work? `cat.entry_part1(part='1')` is probably the simplest form.

**Comment by martindurant** [comment]
Yes, unfortunately the API is still somewhat in flux...

File: docs/source/catalog.rst (hunk 1 above)
Comment (ah-) [reply docs/source/catalog.rst]: That's even nicer, updated.

**Comment by martindurant** [comment]
please merge from master, it will make the failure go away.

**Pull request merged**